code/modules/mob/living/silicon/pai/pai_defense.dm

```diff
@@ -41,34 +40,39 @@
 			Paralyze(200)
 
 /mob/living/silicon/pai/attack_hand(mob/living/carbon/human/user, list/modifiers)
-	if(user.combat_mode)
-		user.do_attack_animation(src)
-		if (user.name == master)
-			visible_message(span_notice("Responding to its master's touch, [src] disengages its holochassis emitter, rapidly losing coherence."))
-			if(do_after(user, 1 SECONDS, TRUE, src))
-				fold_in()
-				if(user.put_in_hands(card))
-					user.visible_message(span_notice("[user] promptly scoops up [user.p_their()] pAI's card."))
-		else
-			visible_message(span_danger("[user] stomps on [src]!."))
-			take_holo_damage(2)
-	else
+	. = FALSE
+	if(SEND_SIGNAL(src, COMSIG_ATOM_ATTACK_HAND, user, modifiers) & COMPONENT_CANCEL_ATTACK_CHAIN)
+		. = TRUE
+	if(!user.combat_mode)
 		visible_message(span_notice("[user] gently pats [src] on the head, eliciting an off-putting buzzing from its holographic field."))
+		return
+	user.do_attack_animation(src)
+	if(user.name != master)
+		visible_message(span_danger("[user] stomps on [src]!."))
+		take_holo_damage(2)
+		return
+	visible_message(span_notice("Responding to its master's touch, [src] disengages its holochassis emitter, rapidly losing coherence."))
+	if(!do_after(user, 1 SECONDS, TRUE, src))
+		return
+	fold_in()
+	if(user.put_in_hands(card))
+		user.visible_message(span_notice("[user] promptly scoops up [user.p_their()] pAI's card."))
 
 /mob/living/silicon/pai/bullet_act(obj/projectile/Proj)
 	if(Proj.stun)
 		fold_in(force = TRUE)
 		src.visible_message(span_warning("The electrically-charged projectile disrupts [src]'s holomatrix, forcing [src] to fold in!"))
 	. = ..(Proj)
 
-/mob/living/silicon/pai/IgniteMob(mob/living/silicon/pai/P)
-	return FALSE //No we're not flammable
+/mob/living/silicon/pai/IgniteMob()
+	fire_stacks = 0
+	. = ..()
 
 /mob/living/silicon/pai/proc/take_holo_damage(amount)
 	emitterhealth = clamp((emitterhealth - amount), -50, emittermaxhealth)
 	if(emitterhealth < 0)
 		fold_in(force = TRUE)
-	if (amount > 0)
+	if(amount > 0)
```

Comment:
not necessary

Reply:
i hate you i hate your existence i wish you were never born and i wish you would burn in hell forever

Reply:
![image](https://user-images.githubusercontent.com/53777086/147632900-ce18791d-36f3-4d6e-a8d1-88e61c1aefc6.png)

Reply:
![2021-12-29_01:31:02_%](https://user-images.githubusercontent.com/16159590/147633615-cf8612e7-8c77-42eb-86ce-221fe4b47b43.png)
sorry i don't make the rules :chad: (chad flag in case you aren't chad enough to understand)

Reply:
i will scrape the walls of the brains of the innocent